modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/infrastructure/snapshot/GlobalSnapshotConsensusStateAdvancer.scala

```diff
@@ -257,6 +261,15 @@ object GlobalSnapshotConsensusStateAdvancer {
                                 .forOrdinal(signedArtifact.ordinal) { implicit hasher =>
                                   for {
                                     hashedSnapshot <- signedArtifact.toHashed
+                                    numBlocks = signedArtifact.blocks.size
+                                    numTransactions = signedArtifact.blocks.map(_.block.transactions.size).sum
+                                    _ <- logger.debug(
+                                      s"Adding GlobalSnapshot to storage for " +
+                                        s"ordinal=${signedArtifact.ordinal.value.value} with hash=${hashedSnapshot.hash.value} " +
+                                        s"numSigners=${signedArtifact.proofs.size} " +
+                                        s"numBlocks=$numBlocks " +
+                                        s"numTransactions=$numTransactions"
+                                    )
                                     _ <- lastNGlobalSnapshotStorage.set(hashedSnapshot, majorityArtifactInfo.context)
```

Comment:
This is also deliberate log statement; it's very useful for gauging exact round change time for measuring other log statements.